lib/bandit/http1/adapter.ex

```diff
@@ -286,30 +326,27 @@ defmodule Bandit.HTTP1.Adapter do
   # Internal Reading
   ##################
 
-  @spec read(ThousandIsland.Socket.t(), non_neg_integer(),
-          read_length: non_neg_integer(),
-          read_timeout: timeout()
-        ) :: {:ok, iolist()} | {:error, :closed | :timeout | :inet.posix()}
-  defp read(socket, to_read, opts \\ []) do
-    read_size = Keyword.get(opts, :read_length, 1_000_000)
-    read_timeout = Keyword.get(opts, :read_timeout)
-    do_read(socket, to_read, [], read_size, read_timeout)
+  @compile {:inline, read_available: 2}
+  @spec read_available(ThousandIsland.Socket.t(), timeout()) ::
+          {:ok, binary()} | {:error, :closed | :timeout | :inet.posix()}
+  defp read_available(socket, read_timeout) do
+    ThousandIsland.Socket.recv(socket, 0, read_timeout)
```

Comment:
This patch does not change the code logic at all - it works the same way it worked before. The only thing which differs is that now it's obvious that `ThousandIsland.Socket.recv(socket, 0, read_timeout)` is invoked in some cases (I've just spotted that `read_timeout` is `nil` in some cases otherwise whatever user provided in `Keyword.get(opts, :read_timeout)`). Before that was hidden in `read/3` function.

Unfortunately the documentation isn't clear on how `0` works with a timeout. But it says that timeout's default values is `:infinity` (not `nil` which we have at the moment, I think it would be beneficial to fix that right away):
```
Receives a packet from a socket in passive mode. A closed socket is indicated by return value {error, closed}.

Argument Length is only meaningful when the socket is in raw mode and denotes the number of bytes to read. If Length is 0, all available bytes are returned. If Length > 0, exactly Length bytes are returned, or an error; possibly discarding less than Length bytes of data when the socket is closed from the other side.

The optional Timeout parameter specifies a time-out in milliseconds. Defaults to infinity.
```

What do you think if we move discussion on these concerns to a new issue and eventually work out a separate PR?

Reply:
Good point. I think this reads a lot better overall, given that it's the same net effect.

Agreed that we should fix that everywhere on a separate PR I think.